Adding a README to the bug-bounty-list directory (#26)
disclose/disclose: bug-bounty-list/README.md

# disclose.io/bug-bounty-list

> > The community-powered database of ALL known public bug bounty and vulnerability programs, including their safe-harbor status.  

Reviewer: ```suggestion
> The community-powered index of ALL known public bug bounty and vulnerability disclosure programs, including their safe harbor status.  
```

## How to contribute  

> If you've found a new program, new information on an existing program, or think a program should be deleted, you can contribute to bug-bounty-list by issuing a pull request (PR) to the repo. If you're new to Github, [this article](https://www.thinkful.com/learn/github-pull-request-tutorial/#Time-to-Submit-Your-First-PR) is a good primer on how PRs work. 

Reviewer: ```suggestion
If you've found a new program, new information on an existing program, or think a program should be deleted, you can contribute to bug-bounty-list by issuing a pull request (PR) to the repo. If you're new to Github, [this article](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) is a good primer on how PRs work. 
```

Would something like this in GitHub documentation work? We may need a third-party resource if it's too complex, but first-party is probably ideal.

Author: It could, yeh. In the meantime my goal is getting a "how-to" out to the community as quickly as we can, but we can definitely extend it in the future.

## Before you go hunting for new programs to add...

Programs on bug-bounty-list need to satisfy the definition of a public bug bounty or vulnerability disclosure program, which means they need two key components:
  1. A publicly accessible vulnerability disclosure policy sometimes called a program brief, or bounty brief, and 
  2. A publicly accessible channel for vulnerability submission **which is explicitly mentioned in the vulnerability disclosure policy**.  
  
Some examples of suggestions that won't be accepted:
  1. Private programs  
  2. Invitation-only programs  
  3. Programs that you heard about from a friend but aren't listed publicly  
  4. Programs without a public policy or a nominated channel for communication  

## Definition of fields 

> **program_name (required):** What is the name of the company or product team responsible for the program?  

Reviewer: ```suggestion
- **program_name (required):** What is the name of the company or product team responsible for the program?  
```

> **policy_url (required):** Where can the policy which defines the program be found? (Usually a webpage e.g. /security, a URL on a vulnerability management platform, or in a robots.txt or security.txt file).  

Reviewer: ```suggestion
- **policy_url (required):** Where can the policy which defines the program be found? (Usually a webpage e.g. /security, a URL on a vulnerability management platform, or in a robots.txt or security.txt file).  
```

> **submission_url (required):** Where does policy say vulnerability submissions should be sent? (Usually a web form, a URL on a vulnerability management platform, or an email address. Emails can be added as URLs using 'mailto:').  

Reviewer: ```suggestion
- **submission_url (required):** Where does policy say vulnerability submissions should be sent? (Usually a web form, a URL on a vulnerability management platform, or an email address. Emails can be added as URLs using 'mailto:').  
```

> **submission_url (required):** Where does policy say vulnerability submissions should be sent? (Usually a web form, a URL on a vulnerability management platform, or an email address. Emails can be added as URLs using 'mailto:').  

Reviewer: I believe the `mailto:` links were removed, so we should decide on a convention and reapply them to all email addresses in `submission_url` fields if deciding to go this route.

Author: Thanks, agree.

> **launch_date (optional):** When does the program say it was launched?  

Reviewer: ```suggestion
- **launch_date (optional):** When does the program say it was launched?  
```

> **cash_reward (optional):** Does the program commit to rewarding vulnerabilities?  

Reviewer: ```suggestion
- **cash_reward (optional):** Does the program commit to rewarding vulnerabilities?  
```

> **swag (optional):** Does the program commit to sending swag out for vulnerabilities?  

Reviewer: ```suggestion
- **swag (optional):** Does the program commit to sending swag out for vulnerabilities?  
```

> **hall_of_fame (optional):** Does the program have a listing or Hall of Fame page acknowledging vulnerability reporters?

Reviewer: ```suggestion
- **hall_of_fame (optional):** Does the program have a listing or Hall of Fame page acknowledging vulnerability reporters?
```

> **hall_of_fame (optional):** Does the program have a listing or Hall of Fame page acknowledging vulnerability reporters . 

Reviewer: ```suggestion
**hall_of_fame (optional):** Does the program have a listing or Hall of Fame page acknowledging vulnerability reporters?
```

> **safe_harbor (required):** Does the program support full or partial safe-harbor?  

Reviewer: ```suggestion
- **safe_harbor (required):** Does the program support full or partial safe-harbor?  
```

> **safe_harbor (required):** Does the program support full or partial safe-harbor?  

Reviewer: You may want to refer to the root directory README, which briefly outlines what "partial" safe harbor means.

Author: Ack. We should extend the definitions in the master docs as well.

~~~~
  {
    "program_name": "Tesla",
    "policy_url": "https://bugcrowd.com/tesla",
    "submission_url": "https://bugcrowd.com/tesla/report",
    "launch_date": "",
    "cash_reward": true,
    "swag": false,
    "hall_of_fame": true,
    "safe_harbor": "full"
  },
~~~~

*Example: Tesla... Can you find the date they first launched their bug bounty program?*

> ## How to fill out the safe-harbor field

Reviewer: ```suggestion
## How to fill out the safe harbor field
```

> The list supports three safe-harbor states: Full, Partial, and None. The flow below will help you determine which applies to the policy as you contribute to the list:  

Reviewer: ```suggestion
The list supports three safe harbor states: Full, Partial, and None. The flow below will help you determine which applies to the policy as you contribute to the list:  
```

> The list supports three safe-harbor states: Full, Partial, and None. The flow below will help you determine which applies to the policy as you contribute to the list:  

Reviewer: If you don't like the changes below, I'd suggest just omitting these steps and linking to https://github.com/disclose/disclose#partial-safe-harbor in the parent readme, which explains the same idea (albeit a bit more briefly).

1. Does the policy mention anything about legal protections for good-faith hackers? If it does, go to Step 2. If not, then you should choose **"None"**. 

> 2. Does the policy talk about authorization of testing and/or exemption from legal terms? If it does, you should choose **"Full"**. If not got to Step 3.

Reviewer: ```suggestion
2. Does the policy talk about authorization of testing and/or an exemption from preexisting legal terms? In addition, does the policy clearly define the scope of testing, the compensation for valid findings, the official communication channels, and a disclosure policy? If so, you should choose **"Full"**. If not got to Step 3.
```

> 2. Does the policy talk about authorization of testing and/or exemption from legal terms? If it does, you should choose **"Full"**. If not got to Step 3.

Reviewer: I guess my previous comment may be irrelevant if you're going to restate these terms here, but I'd use similar language to avoid conflicting definitions (or rewrite the original definition in the parent readme).

"Partial" should be used when the Disclose.io requirements are _not_ met but the program still has a statement about authorizing good-faith testing, and "Full" should only be used if the Disclose.io requirements are met alongside such an authorization statement.

> 3. Does the policy talk about "not legally pursuing researchers" or use similar language, but does not explicitly grant authorization and/or exemption for security testing? If it does, you should choose **"Partial"**. 

Reviewer: ```suggestion
3. Does the policy talk about "not legally pursuing researchers" or use similar language, but does not explicitly grant authorization and/or exemption for security testing? Are any of the key parameters described above ambiguously defined or missing completely? If so, you should choose **"Partial"**. 
```
